sdk/src/Services/S3/Custom/Model/DeleteObjectsRequest.Extensions.cs

```diff
@@ -30,13 +31,69 @@ namespace Amazon.S3.Model
     /// </summary>
     public partial class DeleteObjectsRequest : AmazonWebServiceRequest
     {
+        private List<DeleteError> validationErrors;
+
+        /// <summary>
+        /// Gets the list of keys that were skipped during client-side validation.
+        /// </summary>
+        public List<DeleteError> ValidationErrors
+        {
+            get
+            {
+                if (validationErrors == null)
+                {
+                    validationErrors = new List<DeleteError>();
+                }
+                return validationErrors;
+            }
+        }
+
+        private static Logger Logger
+        {
+            get
+            {
+                return Logger.GetLogger(typeof(DeleteObjectsRequest));
+            }
+        }
 
         /// <summary>
         /// Add a key to the set of keys of objects to be deleted.
         /// </summary>
         /// <param name="key">Object key</param>
+        /// <remarks>
+        /// Invalid keys (null, empty, or longer than 1024 characters) are skipped with an INFO-level log message.
+        /// These validation errors are also tracked in the ValidationErrors collection.
+        /// This allows the DeleteObjects operation to proceed with valid keys even when some keys are invalid.
+        /// </remarks>
         public void AddKey(string key)
         {
+            // Skip invalid keys instead of throwing exceptions
+            // This allows the DeleteObjects operation to proceed with valid keys
+            if (string.IsNullOrEmpty(key))
+            {
+                string message = "Skipping null or empty key in DeleteObjects request";
+                Logger.InfoFormat(message);
+                
+                ValidationErrors.Add(new DeleteError {
+                    Key = key ?? "(null)",
+                    Code = "InvalidKey",
+                    Message = "Key cannot be null or empty"
+                });
+                return;
+            }
+            if (key.Length > 1024)
+            {
+                string message = string.Format("Skipping key in DeleteObjects request: key length {0} exceeds maximum of 1024 characters", key.Length);
+                Logger.InfoFormat(message);
+                
+                ValidationErrors.Add(new DeleteError {
+                    Key = key.Substring(0, Math.Min(key.Length, 128)) + (key.Length > 128 ? "..." : ""),
```

Comment:
Min here is not truly required, also does it make sense to trim it it 128 when 1024 length is legit it can be issue when you have common prefix of 128 chars length